### PR TITLE
YONK-762: Discussions score issue fixed

### PR DESCRIPTION
**Pull request by aamishbaloch**
Discussions scores are now updating for the learners who respond to a post, if user deletes his own actual post or comment.

### CODE REVIEW

**Comment by aamishbaloch** [comment]
@ziafazal Can you review this?

**Comment by aamishbaloch** [comment]
@ziafazal Changes in platforms can be seen in [ee6c70da1624abe91b28612386b56082387253d3](https://github.com/edx-solutions/edx-platform/commit/ee6c70da1624abe91b28612386b56082387253d3) commit.

File: social_engagement/engagement.py [hunk 1]
```diff
@@ -251,3 +253,102 @@ def handle_progress_post_save_signal(sender, instance, **kwargs):
                 # Notifications are never critical, so we don't want to disrupt any
                 # other logic processing. So log and continue.
                 log.exception(ex)
+
+
+def get_involved_users_in_thread(incoming_request, thread_id):
+    from lms.djangoapps.discussion_api.views import CommentViewSet
```
Comment (ziafazal) [inline social_engagement/engagement.py]: Put this after docstring.

File: social_engagement/engagement.py [hunk 2]
```diff
@@ -251,3 +253,101 @@ def handle_progress_post_save_signal(sender, instance, **kwargs):
...
+    for comment in response.data["results"]:
+        users.add(comment["author"])
+        if comment["child_count"] > 0:
+            response_comment_ids.append(comment["id"])
+
+    while response.data["pagination"]["next"]:
+        response_page += 1
+        request.GET.update({"page": response_page})
+
+        response = CommentViewSet().list(request)
```
Comment (ziafazal) [inline social_engagement/engagement.py]: this is duplicate code block it can be removed.

File: social_engagement/engagement.py [hunk 3]
```diff
@@ -251,3 +253,101 @@ def handle_progress_post_save_signal(sender, instance, **kwargs):
...
+    for response_comment_id in response_comment_ids:
+        response = CommentViewSet().retrieve(request, response_comment_id)
+        for comment in response.data["results"]:
+            users.add(comment["author"])
+
+        while response.data["pagination"]["next"]:
+            response_page += 1
+            request.GET.update({"page": response_page})
+
+            response = CommentViewSet().retrieve(request, response_comment_id)
```
Comment (ziafazal) [inline social_engagement/engagement.py]: duplicate code block.

File: social_engagement/engagement.py [hunk 4]
```diff
@@ -251,3 +253,101 @@ def handle_progress_post_save_signal(sender, instance, **kwargs):
...
+        for comment in response.data["results"]:
+            users.add(comment["author"])
+
+    if parent_id:
+        # getting parent of the comment
+        comment = cc.Comment.find(parent_id)
+        users.add(comment.username)
+
+    # getting creator of the post by thread_id
+    response = ThreadViewSet().retrieve(request, thread_id)
```
Comment (ziafazal) [inline social_engagement/engagement.py]: I think this code block should be in separate method.

File: social_engagement/engagement.py [hunk 5]
```diff
@@ -251,3 +253,101 @@ def handle_progress_post_save_signal(sender, instance, **kwargs):
...
+
+    while response.data["pagination"]["next"]:
+        response_page += 1
+        request.GET.update({"page": response_page})
+
+        response = CommentViewSet().retrieve(request, response_comment_id)
+        for comment in response.data["results"]:
+            users.add(comment["author"])
+
+    if parent_id:
```
Comment (ziafazal) [inline social_engagement/engagement.py]: why this needs to be here?

File: social_engagement/handlers.py [hunk 6]
```diff
@@ -27,3 +29,11 @@ def thread_comment_signal_handler(sender, **kwargs):  # pylint: disable=unused-a
     user_id = getattr(post, 'user_id', None)
     if course_id and user_id:
         task_update_user_engagement_score.delay(unicode(course_id), user_id)
+
+    # present only in two cases: thread_deleted & comment_deleted
+    if 'involved_users' in kwargs:
+        involved_users = kwargs['involved_users']
+        users = User.objects.filter(~Q(id=user_id), username__in=involved_users)
```
Comment (ziafazal) [inline social_engagement/handlers.py]: nit: may using this filter is more readable.
```
 users = User.objects.filter(username__in=involved_users).exclude(id=user_id)
```

**Review by ziafazal — CHANGES_REQUESTED** [review]
@aamishbaloch I'm do with first pass of code review.

**Comment by aamishbaloch** [comment]
@ziafazal Done with the changes. Can you review the changes?

**Review by ziafazal — APPROVED** [review]
👍

**Pull request merged**